src/database/MySQLTable.ts

```diff
@@ -25,7 +26,10 @@ export class MySQLTable {
   public async apply(row: ParsedRecord): Promise<void> {
     this.buffer[row.action].push(row);
 
-    if (this.buffer[row.action].length >= this.flushLimit) {
+    if (row.action === RecordAction.DelayedInsert) {
+      this.buffer[RecordAction.Delete].push({ action: RecordAction.Delete, ...row });
+    }
+    else if (this.buffer[row.action].length >= this.flushLimit) {
```

Comment:
I think this `else` relationship deserves a comment if not a separate logic flow to make sure DelayedInsert is not flushed here.